tests/e2e/scenarios/test_reborn_webui_v2_legacy_tool_permissions.py

```
@@ -0,0 +1,297 @@
+    await expect(page.get_by_placeholder("Search settings...")).to_be_visible(timeout=15000)
+    await expect(page.get_by_text("Tool permissions")).to_be_visible(timeout=5000)
+
+    return {
+        "context": context,
+        "page": page,
+        "auto_approve_requests": auto_approve_requests,
+        "permission_requests": permission_requests,
+    }
+
+
+def _tool_row(page, name: str):
```

Ruff (0.15.18)

[warning] Missing return type annotation for private function `_tool_row`

(ANN202)